lib/src/navigation.dart

```diff
@@ -106,6 +108,12 @@ class BottomNavScaffold extends ConsumerWidget {
     final currentTab = ref.watch(currentBottomTabProvider);
     final tabs = ref.watch(tabsProvider);
 
+    bool isOnline = true;
+
+    ref.listen(connectivityChangesProvider, (_, connectivity) {
```

Comment:
You should not use `ref.listen` here but fetch connectivity value directly in `_onItemTapped`.